src/decompose.js

```diff
@@ -0,0 +1,66 @@
+import { scale } from './scale'
+import { compose } from './transform'
+/**
+ * Decompose a matrix into translation, rotation and scaling components, optionally
+ * take horizontal and vertical flip in to consideration.
+ * Note this function decomposes a matrix in rotation -> scale -> translation order. I.e. for
+ * certain translation T {tx, ty}, rotation R and scaling S { sx, sy }, it's only true for:
+ *  decompose(compose(T, S, R)) === { translate: T, rotation: R, scale: S }
+ * composing in a different order may yield a different decomposition result.
+ * @param matrix {Matrix} Affine Matrix
+ * @param  flipX {boolean} Whether the matrix contains vertical flip, i.e. mirrors on x-axis
+ * @param  flipY {boolean} Whether the matrix contains horizontal flip, i.e. mirrors on y-axis
+ * @returns {Transform} A transform object consisted by its translation, rotation
+ * and scaling components.
+ */
+export function decompose (matrix, flipX = false, flipY = false) {
```

Comment:
Do you think that we can detect the fliX or flipY in some way? I was wondering if looking at `a` and `d` items could be enough. Basically a flipped matrix should have negative values.

Reply:
We can only detect if there is `flipX` **or** `flipY`, but not both. If you flip an image both horizontally and vertically, it's identical to that image being rotated by 180 degrees.

Reply:
Ok, clear.
Thanks for the detailed explanation.